### ISSUE TITLE
GooglePlus e-mail exception

### ISSUE THREAD

**Issue by ostracoder** [TerribleDev/OwinOAuthProviders#97]
I have an exception in the GooglePlusAuthenticatedContext constructor when the e-mail cannot be retrieved from the Google+ account. The exception is catched and logged in AuthenticateCoreAsync() but, in the end, this results in a failed authentication.

The problem is on this query:
var email = (from e in person["emails"]
 where e["type"].ToString() == "account"
 select e).FirstOrDefault();

This fails when person["emails"] is null for some reason (probably when the user does not allow sharing of his e-mail).

**Comment by jerriep** [comment]
@daltonwide I strongly suggest switching over to the Microsoft maintained Google provider which is in the `Microsoft.Owin.Security.Google` Nuget package. Can you please switch over and check whether that one works better for you?

**Comment by ostracoder** [comment]
Thanks for your reply.
Yes, the provider from Microsoft works for me but is less powerful as it does not call the Google+ API to retrieve user information.
So, a fix would still be useful...

**Comment by jerriep** [comment]
Yeah, sorry I simply do not have time right now to fix. If you want to fork the project, fix it, and submit a pull request I will be very grateful.

Alternative is to use the MS one and make the call to get the G+ user info yourself. Something like this:

```
app.UseGoogleAuthentication(new GoogleOAuth2AuthenticationOptions
{
    ClientId = "blha",
    ClientSecret = "blah",
    Provider = new GoogleOAuth2AuthenticationProvider
    {
        OnAuthenticated = async context =>
        {
            // Get access token
            string accessToken = context.AccessToken;

            // Call G+ API with HttpClient
            HttpClient httpClient = new HttpClient();
            var response = await httpClient.GetAsync("https://www.googleapis.com/plus/v1/people/me?access_token=" + Uri.EscapeDataString(accessToken), CancellationToken.None);
            response.EnsureSuccessStatusCode();
            var text = await response.Content.ReadAsStringAsync();

            // Extract the value you want here from the text retrieved from G+
        }
    }
});
```

I did not test the above code, but it should work fine...
